### DISCUSSION TITLE
How do I get the realtime subscription clientId on the frontend JS-SDK?

### DISCUSSION THREAD

**Discussion by kidino** [pocketbase/pocketbase#3680] [Q&A]
Just as a test / PoC on the realtime feature, I am going to create collection (say active_users). Then upon the page load, it will immediately subscribe to the active_users collection. I believe with the PB_CONNECT eventsource by the Pb client, we will receive a clientId value.

On the frontend, I will also send an insert to the active_users with the clientId value.

I will have kinda like an avatar on the screen representing the current user based on the clientId. And everytime the user hit up, down, left right -- the avatar will move. This new location will also be sent to the backend by updating the active_users collection.

The final outcome is where I have a screen with random avatar representation of people who are live / active on the website.

I also plan to implement onRealtimeDisconnectRequest event on the backend to remove records from active_users when people disonnects.

How issue right now, is I don't know how to get the realtime subscription clientId that we get from PB_CONNECT

**Comment by ganigeorgiev** [comment]
I'm not sure that I understand the use case, but if you use the JS SDK you should be able to access the clientId by calling `pb.realtime.clientId`.

**Reply by kidino** [reply]
Thank you. Got it. I had `console.log( pb.realtime.clientId )` immediately after declaring the `pb` object. And I got an empty string. But I do get the clientId value when I wait 1 second, or get it with a button trigger.

Is there like a callback that I should properly be using to make sure that the clientId is actually loaded?

**Reply by ganigeorgiev** [reply]
There is no "connect" hook in the JS SDK at the moment (_I've added it in my todo to consider propagating the `PB_CONNECT` event to the initial subscriptions_) but `pb.realtime.clientId` will be always set after `await`-ing the subscribe call.

But again, I don't fully understand your use case so maybe there is a better solution for what you are trying to do (_eg. with the [`onRealtimeConnectRequest` hook](https://pocketbase.io/docs/js-event-hooks/#onrealtimeconnectrequest)_).

**Reply by kidino** [reply]
Thanks. That's OK. I can work with what Pb has for now. Perhaps I will show my demo later.
